inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,249 @@
-
-        if has_pub_info and has_page_or_artid:
-            # NOTE: Only peer reviewed journals should have this collection
-            # we are adding it here but ideally should be manually added
-            # by a curator.
-            data['collections'].append({'primary': "Published"})
-            # Add Citeable collection if not present
-            collections = [x['primary'] for x in data['collections']]
-            if "Citeable" not in collections:
-                data['collections'].append({'primary': "Citeable"})
-    # ============================
-    # Title source and cleanup
-    # ============================
-    try:
-        # Clean up all extra spaces in title
-        data['titles'][0]['title'] = " ".join(
-            data['titles'][0]['title'].split()
+    builder = LiteratureBuilder(source='submitter')
+
+    for author in form_fields.get('authors', []):
+        builder.add_author(builder.make_author(
+            author['full_name'],
+            affiliations=force_force_list(author['affiliation']),
+        ))
+
+    for supervisor in form_fields.get('supervisors', []):
+        builder.add_author(builder.make_author(
+            supervisor['full_name'],
+            affiliations=force_force_list(supervisor['affiliation']),
+            supervisor=True
+        ))
+
+    builder.add_title(title=form_fields.get('title', []))
+
+    document_type = 'conference paper' if form_fields.get('conf_name') \
+        else form_fields.get('type_of_doc', [])
+
+    builder.add_if_non_empty(
+        'document_type',
+        document_type=document_type
+    )
+
+    builder.add_if_non_empty(
+        'abstract',
+        abstract=form_fields.get('abstract'),
+        source='arXiv' if form_fields.get('categories') else None
+    )
+
+    if form_fields.get('arxiv_id') and form_fields.get('categories'):
+        arxiv_categories = _filter_arxiv_categories(
+            form_fields.get('categories'),
+            form_fields.get('arxiv_id')
         )
-        title = data['titles'][0]['title']
-    except (KeyError, IndexError):
-        title = ""
-    if form_fields.get('title_arXiv'):
-        title_arxiv = " ".join(form_fields.get('title_arXiv').split())
-        if title == title_arxiv:
-            data['titles'][0]["source"] = "arXiv"
-        else:
-            data['titles'].append({
-                'title': title_arxiv,
-                'source': "arXiv"
-            })
-    if form_fields.get('title_crossref'):
-        title_crossref = " ".join(
-            form_fields.get('title_crossref').split()
+
+        builder.add_arxiv_eprint(
+            arxiv_id=form_fields.get('arxiv_id'),
+            arxiv_categories=arxiv_categories
         )
-        if title == title_crossref:
-            data['titles'][0]["source"] = "CrossRef"
-        else:
-            data['titles'].append({
-                'title': title_crossref,
-                'source': "CrossRef"
-            })
+
+    builder.add_if_non_empty(
```

Comment:
As mentioned above, using everywhere `builder.add_if_non_empty` defeats a bit the purpose of having the builder, you should have methods in the builder to build these fields (and avoid passing empty values here)